### DISCUSSION TITLE
FzfLua changes editor colors

### DISCUSSION THREAD

**Discussion by stounfo** [ibhagwan/fzf-lua#1606] [Q&A] [answered]
I use nvim with termgutcolors = false
When I run :FzfLua it changes the fg colors of my editor

Without FzfLua:
<img width="1512" alt="Screenshot 2024-12-20 at 12 28 02 AM" src="https://github.com/user-attachments/assets/e284139f-2685-4645-a782-34b5e145bd4d" />
With FzfLua:
<img width="1512" alt="Screenshot 2024-12-20 at 12 28 16 AM" src="https://github.com/user-attachments/assets/644f6a78-48ec-4347-83c1-7cf28acd2e59" />
Does anyone know what might be causing this issue?

**Comment by ibhagwan** [comment]
> Does anyone know what might be causing this issue?

Does this happen with `mini.sh`?

**Reply by stounfo** [reply]
<img width="1512" alt="Screenshot 2024-12-20 at 12 35 35 AM" src="https://github.com/user-attachments/assets/92748b2a-ec20-4592-8a63-faf6c2733b96" />
nope

**Reply by stounfo** [reply]
I tried to reproduce the issue with mini.sh but I couldn’t.

If this isn’t a common issue, I will try again tomorrow

**Reply by ibhagwan** [reply]
> I tried to reproduce the issue with mini.sh but I couldn’t.
> 
> If this isn’t a common issue, I will try again tomorrow

Alright, keep me posted if you need my help.

**Reply by stounfo** [reply]
this config reproduce the issue for me
The StatusLine changes when I run :FzfLua.
```lua
vim.env.LAZY_STDPATH = ".repro"
load(
    vim.fn.system(
        "curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"
    )
)()

require("lazy.minit").repro({
    spec = {
        "ibhagwan/fzf-lua"
    },
})

local buf = vim.api.nvim_create_buf(false, true)
local content = {
    "def example():",
    "    pass",
}

vim.api.nvim_buf_set_lines(buf, 0, -1, false, content)
vim.api.nvim_buf_set_option(buf, "filetype", "python")

vim.api.nvim_set_current_buf(buf)

vim.opt.termguicolors = false
vim.cmd("hi StatusLine ctermbg=1")
```

os: MacOs 15.2
nvim: v0.10.2
fzf: 0.57.0
Tested in any shell, terminal emulator, and with default fzf settings.

**Reply by ibhagwan** [reply, marked as answer]
> this config reproduce the issue for me
> The StatusLine changes when I run :FzfLua.

That's not a bug, there's a different highlight group when the statusline is not active `StatusLineNC`, if you run `:Lazy` you'll see the same effect (statusline color will change when the float win is open).

To "solve" this, set the hlgroup colors to same values as `StatusLine`  (using the colors from the repro):
```vim
hi! StatusLineNC ctermbg=1 guifg=#828bb8 guibg=#1e2030
```

Now regarding the effect seen in the OP, not sure why that would be the effect in your setup but might be related to the backdrop (shading) of the background window, try running without backdrop:
> Try different values
```lua
:FzfLua files winopts.backdrop=false    -- disabled, no backdrop
:FzfLua files winopts.backdrop=20
:FzfLua files winopts.backdrop=60       -- FzfLua's default, not sure what's Lazy's default
:FzfLua files winopts.backdrop=100      -- also disabled
```

**Reply by stounfo** [reply]
backdrop = false resolves the issues for me, thanks for the help! And for a great plugin!

But just in case
When running :Lazy and :FzfLua with the configuration below, the background buffer displays different colors
## :Lazy
<img width="1419" alt="Screenshot 2024-12-20 at 4 39 29 PM" src="https://github.com/user-attachments/assets/cc24a830-fff0-4120-8e86-232f59be85db" />

## :FzfLua
<img width="1243" alt="Screenshot 2024-12-20 at 4 39 41 PM" src="https://github.com/user-attachments/assets/2b78c498-eea7-44af-a59f-b1bcfe592fe9" />

It seems like a bug to me, but maybe I just don't understand nvim highlighting


```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
	spec = {
		{
			"ibhagwan/fzf-lua",
			config = function()
				require("fzf-lua").setup({
					fzf_colors = false,
					winopts = {
						width = 0.9,
						preview = {
							layout = "vertical",
						},
					},
				})
			end,
			lazy = false,
		},
	},
})

local buf = vim.api.nvim_create_buf(false, true)
local content = {
	"function example()",
	"end",
}

vim.api.nvim_buf_set_lines(buf, 0, -1, false, content)
vim.api.nvim_buf_set_option(buf, "filetype", "lua")

vim.api.nvim_set_current_buf(buf)

vim.opt.termguicolors = false

local set_hl = function(groups)
	for group, settings in pairs(groups) do
		vim.api.nvim_set_hl(0, group, settings)
	end
end

vim.cmd("hi Normal ctermfg=2 ctermbg=0")
vim.cmd("hi NormalNC ctermfg=1 ctermbg=0")
```

**Reply by ibhagwan** [reply]
> It seems like a bug to me, but maybe I just don't understand nvim highlighting

Not a bug, lazy uses a different value for default backdrop, if you play around with different backdrop values you’ll see the difference, the reason it’s more noticeable is because you’re using notermguicolors so each minor backdrop change probably results in a significantly different color as there are only 256 colors to play with.
